src/zenml/utils/source_utils.py

```diff
@@ -98,15 +96,7 @@ def is_third_party_module(file_path: str) -> bool:
         `True` if the file belongs to a third party package, else `False`.
     """
     absolute_file_path = pathlib.Path(file_path).resolve()
-
-    for path in site.getsitepackages() + [
-        site.getusersitepackages(),
-        get_python_lib(standard_lib=True),
-    ]:
-        if pathlib.Path(path).resolve() in absolute_file_path.parents:
-            return True
-
-    return False
+    return get_source_root_path() not in absolute_file_path.parents
```

Comment:
get_source_root_path returns a str. However, a casting to Path would be necessary as the current implementation  seems to not work as expected:

```python
'/home/zenml/tests/unit/utils' in Path('/home/zenml/src/zenml/utils/source_utils.py').parents

False
```

I'm also not sure this work as expected in general. The root repo might very well be at the same level as the venv:

|-.zen
|-venv
|user_src

As such this will make all packages in venv non-third party packages:

e.g.:
```python
`/home/apenner/PycharmProjects/zenml` not in `/home/apenner/PycharmProjects/zenml/venv/lib/python3.8/site-packages/_pytest/cacheprovider.py`

False
```